test/bats/azure-test.bats

```diff
@@ -318,3 +318,47 @@ SLEEP_TIME=1
     result=$(kubectl get pod mutate-demo --namespace default -o json | jq -r ".spec.containers[0].image" | grep @sha)
     assert_mutate_success
 }
+
+@test "validate refresher reconcile count" {
+    sed -i -e "s/keymanagementprovider-akv/kmp-akv-refresh/" \
+        -e "s/1m/1s/" \
+        -e "s/yourCertName/${NOTATION_PEM_NAME}/" \
+        -e '/version: yourCertVersion/d' \
+        -e "s|https://yourkeyvault.vault.azure.net/|${VAULT_URI}|" \
+        -e "s/tenantID:/tenantID: ${TENANT_ID}/" \
+        -e "s/clientID:/clientID: ${IDENTITY_CLIENT_ID}/" \
+        ./config/samples/clustered/kmp/config_v1beta1_keymanagementprovider_akv_refresh_enabled.yaml
+    run kubectl apply -f ./config/samples/clustered/kmp/config_v1beta1_keymanagementprovider_akv_refresh_enabled.yaml
+    assert_success
+    sleep 10
+    count=$(kubectl logs deployment/ratify -n gatekeeper-system | grep "Reconciled KeyManagementProvider" | wc -l)
+    [ $count -ge 4 ]
+}
+
+@test "validate certificate version update" {
+    result=$(kubectl get keymanagementprovider kmp-akv-refresh -o jsonpath='{.status.properties.Certificates[0].Version}')
+    az keyvault certificate get-default-policy -o json >>policy.json
+    wait_for_process 20 10 "az keyvault certificate create --vault-name $KEYVAULT_NAME --name $NOTATION_PEM_NAME --policy @policy.json"
+    sleep 15
+    run rm policy.json
+    refreshResult=$(kubectl get keymanagementprovider kmp-akv-refresh -o jsonpath='{.status.properties.Certificates[0].Version}')
+    [ "$result" != "$refreshResult" ]
```

Comment:
wonder the difference between it and 
```
run []
assert_success
```

Reply:
`result` is the certificate version pulled from the kmp resources at the start of the test, which is the old version of the certificate. And `refreshResult` is the version pulled after a new version of the certificate has been created`. I then use `!=` ensure the version are not equal because if they are then the refresher didn't work. 

My understanding that `run` and `assert_success` would only validate that the command ran successfully, which it would no matter what, as long as the resource existed. Please lmk if my understanding is incorrect and if I could use that instead. :)

Reply:
the `assert_success` checks the exit status, 0 if the `run` command holds true and 1 otherwise. And `assert_success` would return 1 to stop executing the test. Wonder if `[ "$result" != "$refreshResult" ]` behaves the same, as long as it will fail early we can keep it.

Reply:
That's a good question, I don't have the answer too. :) I'm not to familiar with bats.